### PR TITLE
refactor harmonyCompatibilityDependency to es6

### PR DESCRIPTION
**Pull request by timse**
**What kind of change does this PR introduce?**

refactor

**Did you add tests for your changes?**

current tests should pass

**If relevant, link to documentation update:**

N/A

**Summary**

upgrade HarmonyCompatibilityDependency and HarmonyCompatibilityDependencyTemplate to es6

**Does this PR introduce a breaking change?**

No

### CODE REVIEW

File: lib/dependencies/HarmonyCompatiblilityDependency.js [hunk 1]
```diff
@@ -2,24 +2,29 @@
...
-	var usedExports = dep.originModule.usedExports;
-	if(usedExports && !Array.isArray(usedExports)) {
-		var content = "Object.defineProperty(exports, \"__esModule\", { value: true });\n";
-		source.insert(-1, content);
+HarmonyCompatiblilityDependency.Template = class HarmonyExportDependencyTemplate {
+	apply(dep, source) {
+		const usedExports = dep.originModule.usedExports;
+		if(usedExports && !Array.isArray(usedExports)) {
+			const content = `Object.defineProperty(exports, "__esModule", { value: true });
+`;
```
Comment (Kovensky) [inline lib/dependencies/HarmonyCompatiblilityDependency.js]: Not sure on embedded newline vs `\n`

Comment (timse) [inline lib/dependencies/HarmonyCompatiblilityDependency.js]: yeah, should probably switch that back, escaping the `"` felt weird though.

Comment (TheLarkInn) [inline lib/dependencies/HarmonyCompatiblilityDependency.js]: I'll wait on this one before merging. Nice work overall.

Comment (Kovensky) [inline lib/dependencies/HarmonyCompatiblilityDependency.js]: @TheLarkInn wait on what; moving the backtick to the previous line with `\n`?

Comment (TheLarkInn) [inline lib/dependencies/HarmonyCompatiblilityDependency.js]: Yeah. It feels a bit awkward but if it's okay we can :shipit:

**Comment by TheLarkInn** [comment]
This looks like an appveyor flake to me. I'm going to merge. Thank you.

**Pull request merged**